File: core/src/main/java/com/northstar/core/note/NoteDetail.java
package com.northstar.core.note;

import jakarta.validation.constraints.NotNull;
import java.time.Instant;
import java.util.List;
import java.util.UUID;

/**
 * A full note for the reading view: body plus its folder, tags, outgoing links
 * and backlinks. {@code version} is echoed back on update for optimistic
 * locking. The {@code @NotNull} marks exist for the OpenAPI contract — they
 * make the generated client fields required instead of all-optional.
 */
public record NoteDetail(
        @NotNull UUID id,
        @NotNull String title,
        @NotNull String slug,
        @NotNull String folderPath,
        @NotNull String contentMarkdown,
        @NotNull List<String> tags,
        @NotNull Instant createdAt,
        @NotNull Instant updatedAt,
        long version,
        @NotNull List<NoteRef> outgoingLinks,
        @NotNull List<NoteRef> backlinks) {
}

File: mobile/lib/domain/models/note_detail.dart
class NoteDetail {
  const NoteDetail({
    required this.id,
    required this.title,
    required this.slug,
    required this.folderPath,
    required this.contentMarkdown,
    required this.tags,
    required this.status,
    required this.updatedAt,
  });

  final String id;
  final String title;
  final String slug;
  final String folderPath;
  final String contentMarkdown;
  final List<String> tags;
  final String status;
  final DateTime updatedAt;
}
